### PR TITLE
relatively minor edits

### PR DESCRIPTION
**Pull request by phrawzty**
Hello all,

I humbly propose the following PR with some minor edits, notably concerning the following items:

**`README.md`**
* Uses more idiomatic Markdown.
* Adds / clarifies a few points.

**`SOURCES/consul.json` & `SOURCES/consul-ui.json`**
* Added some sample configs; these are packaged with the `-dist` suffix, which is more idiomatic for packaging purposes.

**`SOURCES/sysconfig/consul`**
* Removed some runtime assumptions. This configuration is as minimal as possible - it should be up to the user to decide how to manage their own configuration.

**`SPECS/consul.spec`**
* Moved the UI to `/usr/share/consul-ui/` as specified by the [Fedora packaging guidelines](https://fedoraproject.org/wiki/Packaging:Guidelines#Web_Applications).
* Added the sample config files.
* Leveraged `setup` to unpack the UI archive (instead of running `unzip` manually) as this is both more portable and more idiomatic.
* Bumped the release.

@duritong @tomhillable `r?`

### CODE REVIEW

**Comment by duritong** [comment]
@phrawzty thanks a lot for these suggestions. I have some comments on some of your proposed changes:

* I'm fine with moving the ui dir, however I would assume that installing the ui package should also activate it. But by just deploying a `-dist` config file , the ui isn't activate it at all. It's clear to me that a package should not expose it to the public, but if activating the ui is as trivial as dropping a .json file in the config-dir, then why not do it? Especially, as the http-api is not public by default. And by explicitly having a `%config(noreplace)` for that file, users can always change that. If you don't want the ui, simply remove the package. If you want the ui somewhere else, why deploy it with by a package anyways?

* You are moving the config dir in the package to /etc/consul, while keeping /etc/consul.d in /etc/sysconfig. I'm fine with moving, but the current spec file doesn't build a working package anymore. Actualy /etc/consul is also what the puppet-consul module assumes, hence I would be fine with that.

* consul barfs about not setting `GOMAXPROCS` or setting `GOMAXPROCS=1`, either we don't set it, or set it to a value where we don't get a notice from consul. I don't think a package should set a value, while a service will still warn about that value. If it warns by default and the package doesn't set the option, then it's another question. I would also be fine by just commenting it out, so to have it there for documentation purposes but leave it in the default behavior.

* I already argued about having a working config in #4 and for me a working config includes a data-dir. With your suggested changes, we don't have a data-dir set anymore and imho the package is non-functional. But we still package the data-dir, what for?

So to summarize, I'm fine with:

* moving the ui-dir
* moving the config dir, if it doesn't break the package. Though, this is a change that might break updates. We're probably early enough to still do that.
* having a `config.json-dist`, as well as the ui.json as SOURCES, but the ui.json should not be deployed as `-dist`
* README changes
* unpacking the ui in the setup process (I simply didn't know how to do that... ;-)

I don't agree with changes that make the package (from my point of view) non-functional.

I hope I could express my view clearly enough, otherwise just ask! Also, I'm totally open for discussion, all I did was bringing my arguments to the table.

**Comment by phrawzty** [comment]
Excellent observations! :smile: 

* The confusion regarding `/etc/consul[.d]/` was a mistake on my part - I have fixed it in `sysconfig`.
* Let's split the difference on `GOMAXPROCS` and comment it out - I added an item about this in the README.
* Your arguments concerning the UI config and `data-dir` declaration are sound - I've revised these elements as well.

**Comment by duritong** [comment]
* The ui config file should still get a `%config(noreplace)`, this makes it possible for ppl to edit it
* for share the `%{_datadir}` could be used http://docs.fedoraproject.org/ro/Fedora_Draft_Documentation/0.1/html/RPM_Guide/ch09s07.html

**Comment by phrawzty** [comment]
Great, thanks for the feedback. Looks like we've got a solid SPEC here now. :smile:

**Comment by duritong** [comment]
@dan cool, thanks a lot :smile:
